projects/plugins/jetpack/class.jetpack-gutenberg.php

```diff
@@ -1045,27 +1045,40 @@ public static function notice( $message, $status = 'info', $classes = '' ) {
 	 * @param string $slug Slug of the block.
 	 */
 	public static function set_availability_for_plan( $slug ) {
-		$is_available = true;
-		$plan         = '';
-		$slug         = self::remove_extension_prefix( $slug );
-
-		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
-			if ( ! class_exists( 'Store_Product_List' ) ) {
-				require WP_CONTENT_DIR . '/admin-plugins/wpcom-billing/store-product-list.php';
+		$is_available   = true;
+		$plan           = '';
+		$slug           = self::remove_extension_prefix( $slug );
+		$features_data  = array();
+		$is_simple_site = defined( 'IS_WPCOM' ) && IS_WPCOM;
+		$is_atomic_site = jetpack_is_atomic_site();
+
+		// Check feature availability for Simple and Atomic sites.
+		if ( $is_simple_site || $is_atomic_site ) {
+
+			// Simple sites.
+			if ( $is_simple_site ) {
+				if ( ! class_exists( 'Store_Product_List' ) ) {
+					require WP_CONTENT_DIR . '/admin-plugins/wpcom-billing/store-product-list.php';
+				}
+				$features_data = Store_Product_List::get_site_specific_features_data();
+			} else {
+				// Atomic sites.
+				$option = get_option( 'jetpack_active_plan' );
+				if ( ! empty( $option ) && array_key_exists( 'features', $option ) ) {
+					$features_data = $option['features'];
+				}
 			}
-			$features_data = Store_Product_List::get_site_specific_features_data();
-			$is_available  = in_array( $slug, $features_data['active'], true );
+
+			$is_available = in_array( $slug, $features_data['active'], true );
```

Comment:
Would this throw a PHP notice if the `get_option` for `jetpack_active_plan` fails since there will be no `active` key in the `$features_data` var?

Reply:
Probably yes.  Good call. Something that `array_key_exists()` will handle properly I think. Thanks 🙇‍♂️

Reply:
addressed here https://github.com/Automattic/jetpack/pull/19762/commits/6616bfc8d65ceef406a8b1736f950d5987b286cd